concent_api/core/message_handlers.py

```diff
@@ -1312,27 +1317,42 @@ def handle_send_subtask_results_verify(
             reason=message.concents.ServiceRefused.REASON.InvalidRequest,
         )
 
-    (requestor_has_enough_deposit, provider_has_enough_deposit) = bankster.claim_deposit(
-        concent_use_case=ConcentUseCase.ADDITIONAL_VERIFICATION,
-        requestor_ethereum_address=task_to_compute.requestor_ethereum_address,
-        provider_ethereum_address=task_to_compute.provider_ethereum_address,
-        subtask_cost=task_to_compute.price,
-    )
-
-    if not requestor_has_enough_deposit:
-        return message.concents.ServiceRefused(
-            reason=message.concents.ServiceRefused.REASON.TooSmallRequestorDeposit,
-        )
-    if not provider_has_enough_deposit:
-        return message.concents.ServiceRefused(
-            reason=message.concents.ServiceRefused.REASON.TooSmallProviderDeposit,
-        )
-
     with transaction.atomic(using='control'):
         subtask = get_one_or_none(
             Subtask.objects.select_for_update(),
             subtask_id=compute_task_def['subtask_id'],
         )
+        if subtask is not None and subtask.state_enum in [
+            Subtask.SubtaskState.VERIFICATION_FILE_TRANSFER,
+            Subtask.SubtaskState.ADDITIONAL_VERIFICATION,
+        ]:
+            return message.concents.ServiceRefused(
+                reason=message.concents.ServiceRefused.REASON.DuplicateRequest,
+            )
+
+        (claim_against_requestor, claim_against_provider) = bankster.claim_deposit(
+            subtask_id=task_to_compute.subtask_id,
+            concent_use_case=ConcentUseCase.ADDITIONAL_VERIFICATION,
+            requestor_ethereum_address=task_to_compute.requestor_ethereum_address,
+            provider_ethereum_address=task_to_compute.provider_ethereum_address,
+            subtask_cost=task_to_compute.price,
+            requestor_public_key=requestor_public_key,
+            provider_public_key=provider_public_key,
+        )
+
+        if claim_against_requestor is None:
+            if claim_against_provider is not None:
+                claim_against_provider.delete()
```

Comment:
Please use `discard_claim()`. All operations on claims (and especially deleting them) must be performed inside a critical section.

Reply:
This is not what the blueprint says

> Concent Core is responsible for removing DepositClaim objects if it fails at any point after the operation if the operation has succeeded.
> Bankster removes DepositClaim objects only after they're successfully paid for.

From what I understand in this part, using `discard_claim()`, which is part of the bankster, should be used only after they're successfully paid for.

Reply:
:+1: done